src/core/methods/trackTransactions/trackTransactions.ts

```diff
@@ -16,44 +18,109 @@ export async function trackTransactions() {
   const pollingInterval = getPollingInterval();
   // eslint-disable-next-line no-undef
   let pollingIntervalTimer: NodeJS.Timeout | null = null;
+  // eslint-disable-next-line no-undef
+  let websocketStatusCheckTimer: NodeJS.Timeout | null = null;
   let timestamp = websocketEventSelector(store.getState())?.timestamp;
 
-  const isWebsocketCompleted =
-    websocketConnection.status === WebsocketConnectionStatusEnum.COMPLETED;
-
   const recheckStatus = () => {
     checkTransactionStatus();
   };
 
-  recheckStatus();
+  const startPolling = () => {
+    if (!pollingIntervalTimer) {
+      pollingIntervalTimer = setInterval(recheckStatus, pollingInterval);
+    }
+  };
 
-  if (isWebsocketCompleted) {
-    // Do not set polling interval if websocket is complete
+  const stopPolling = () => {
     if (pollingIntervalTimer) {
       clearInterval(pollingIntervalTimer);
       pollingIntervalTimer = null;
     }
-    store.subscribe(async ({ account: { websocketEvent } }) => {
-      if (websocketEvent?.message && timestamp !== websocketEvent.timestamp) {
-        timestamp = websocketEvent.timestamp;
-        recheckStatus();
+  };
+
+  const setupWebSocketTracking = () => {
+    stopPolling();
+    const unsubscribeWebsocketEvent = store.subscribe(
+      ({ account: { websocketEvent } }) => {
+        if (websocketEvent?.message && timestamp !== websocketEvent.timestamp) {
+          timestamp = websocketEvent.timestamp;
+          recheckStatus();
+        }
       }
-    });
-  } else {
-    // Set polling interval if websocket is not complete and no existing interval is set
-    if (!pollingIntervalTimer) {
-      pollingIntervalTimer = setInterval(recheckStatus, pollingInterval);
+    );
+
+    subscriptions.set(
+      SubscriptionsEnum.websocketEvent,
+      unsubscribeWebsocketEvent
+    );
+  };
+
+  const startWatchingWebsocketStatus = () => {
+    if (
+      websocketConnection.status !==
+        WebsocketConnectionStatusEnum.NOT_INITIALIZED ||
+      websocketStatusCheckTimer
+    ) {
+      return;
     }
-  }
+
+    websocketStatusCheckTimer = setInterval(() => {
+      if (
+        websocketConnection.status === WebsocketConnectionStatusEnum.COMPLETED
+      ) {
+        clearInterval(websocketStatusCheckTimer!);
+        websocketStatusCheckTimer = null;
+        setupWebSocketTracking();
+      }
+    }, 1000);
+  };
+
+  // Initial execution
+  recheckStatus();
 
   function cleanup() {
-    if (pollingIntervalTimer) {
-      clearInterval(pollingIntervalTimer);
-      pollingIntervalTimer = null;
+    stopPolling();
+    if (websocketStatusCheckTimer) {
+      clearInterval(websocketStatusCheckTimer);
+      websocketStatusCheckTimer = null;
     }
   }
 
-  return {
-    cleanup
-  };
+  const unsubscribeWebsocketStatus = store.subscribe(
+    ({ account: { websocketStatus, address } }, prevState) => {
+      const hasStatusChange =
+        prevState.account.websocketStatus !== websocketStatus;
```

Comment:
I think websocketStatus should sit under `config` slice, since it't not account data related, but app config related